src/Symfony/Component/Config/Resource/ClassExistenceResource.php

```diff
@@ -121,20 +130,38 @@ public function unserialize($serialized)
      *
      * @internal
      */
-    public static function throwOnRequiredClass($class)
+    public static function _throwOnRequiredClass($class)
     {
         if (self::$autoloadedClass === $class) {
             return;
         }
+
+        self::throwOnRequiredClass($class, null);
+    }
+
+    /**
+     * @throws \ReflectionException When $class is not found and is required
+     *
+     * @internal
+     */
+    public static function throwOnRequiredClass($class, \Exception $previous = null)
+    {
+        if (class_exists($class, false) || interface_exists($class, false) || trait_exists($class, false)) {
+            return;
+        }
+
         $e = new \ReflectionException("Class $class not found");
```

Comment:
should we add $previous as 3rd arg?

Reply:
Yes, let's add it since it adds information!